pkg/config/cli.go

```diff
@@ -195,11 +197,26 @@ func FileInputFlag() *pflag.FlagSet {
 func GetCmdOpts() CLIOptions {
 	K0sVars = constant.GetConfig(DataDir)
 
+	if controllerOpts.SingleNode {
+		controllerOpts.EnableWorker = true
+		K0sVars.DefaultStorageType = "kine"
+	}
+
+	// when a custom config file is used, verify that it can be opened
+	if CfgFile != constant.K0sConfigPathDefault {
```

Comment:
What happens now if I give `--config /etc/k0s/k0s.yaml` and it does not exist? I'd expect this to error out, but reading the code I'd assume it'll go and use defaults?

Reply:
technically you're right, but it would be a strange behaviour, since `--config` defaults to `/etc/k0s/k0s.yaml` anyway. If we error this out, then running k0s without any config would fail. The idea behind this was to enable backwards compatibility but also allow users to place their config in a default location without having to specify it explicitly.

Reply:
For these you would need some mechanism to detect if `--config` was given or defaulted.

Reply:
Fixed in the latest commit